### ISSUE TITLE
i3d pretrained model

### ISSUE THREAD

**Issue by marsplant** [seominseok0429/Real-world-Anomaly-Detection-in-Surveillance-Videos-pytorch#2]
thanks for your great work.

would you share your i3d pretrained model to extract the video clip feature ?

**Comment by seominseok0429** [comment]
I'll share. Sorry for the late reply.

**Comment by abhishekaich27** [comment]
@seominseok0429 it would be great if you could provide the model and script to extract i3D features.

**Comment by t-v-k-git** [comment]
please provide the scrip to extract I3d features

**Comment by seominseok0429** [comment]
Ours full code available at [CODE](https://drive.google.com/file/d/1DMJR4cR97FzDXfmfPgg6LKeE7LT7x79d/view?usp=sharing)

**Comment by coranholmes** [comment]
> Ours full code available at [CODE](https://drive.google.com/file/d/1DMJR4cR97FzDXfmfPgg6LKeE7LT7x79d/view?usp=sharing)

Hi, thank you for sharing the full codes. But it seems the feature extractor is still not included?

**Comment by seominseok0429** [comment]
Oops. Model is missing. I'll upload it again

**Comment by coranholmes** [comment]
> Oops. Model is missing. I'll upload it again

Please let me know if you have uploaded the latest version. Thank you so much!

**Comment by coranholmes** [comment]
> @coranholmes This is the modified download link. https://drive.google.com/file/d/1xYsBiCSmXjE0BwoiH_Bcm4AWaA1pESHF/view?usp=sharing.
> 
> Thank!

Hi, thank you for the quick response. I have downloaded the new version, but it seems the same as the previous version. Would you please let me know which script is used to extract the features? Thank you.

**Comment by zimengxueying** [comment]
> 
Have you solved the problem? I'm also confused about it. thx

> > @coranholmes This is the modified download link. https://drive.google.com/file/d/1xYsBiCSmXjE0BwoiH_Bcm4AWaA1pESHF/view?usp=sharing.
> > Thank!
> 
> Hi, thank you for the quick response. I have downloaded the new version, but it seems the same as the previous version. Would you please let me know which script is used to extract the features? Thank you.

**Comment by coranholmes** [comment]
> > 
> 
> Have you solved the problem? I'm also confused about it. thx
> 
> > > @coranholmes This is the modified download link. https://drive.google.com/file/d/1xYsBiCSmXjE0BwoiH_Bcm4AWaA1pESHF/view?usp=sharing.
> > > Thank!
> > 
> > 
> > Hi, thank you for the quick response. I have downloaded the new version, but it seems the same as the previous version. Would you please let me know which script is used to extract the features? Thank you.

Well no... But you can consider trying this [repo](https://github.com/GowthamGottimukkala/I3D_Feature_Extraction_resnet) if you just want the i3d extractor.

**Comment by zimengxueying** [comment]
> > > 
> > 
> > 
> > Have you solved the problem? I'm also confused about it. thx
> > > > @coranholmes This is the modified download link. https://drive.google.com/file/d/1xYsBiCSmXjE0BwoiH_Bcm4AWaA1pESHF/view?usp=sharing.
> > > > Thank!
> > > 
> > > 
> > > Hi, thank you for the quick response. I have downloaded the new version, but it seems the same as the previous version. Would you please let me know which script is used to extract the features? Thank you.
> 
> Well no... But you can consider trying this [repo](https://github.com/GowthamGottimukkala/I3D_Feature_Extraction_resnet) if you just want the i3d extractor.

Thanks a lot! I will try it!

**Comment by zimengxueying** [comment]
> > > 
> > 
> > 
> > Have you solved the problem? I'm also confused about it. thx
> > > > @coranholmes This is the modified download link. https://drive.google.com/file/d/1xYsBiCSmXjE0BwoiH_Bcm4AWaA1pESHF/view?usp=sharing.
> > > > Thank!
> > > 
> > > 
> > > Hi, thank you for the quick response. I have downloaded the new version, but it seems the same as the previous version. Would you please let me know which script is used to extract the features? Thank you.
> 
> Well no... But you can consider trying this [repo](https://github.com/GowthamGottimukkala/I3D_Feature_Extraction_resnet) if you just want the i3d extractor.

Another question, the original paper used i3d two-stream, but the link is resnet, is the speed and accuracy a little different right?

**Comment by coranholmes** [comment]
> > > > 
> > > 
> > > 
> > > Have you solved the problem? I'm also confused about it. thx
> > > > > @coranholmes This is the modified download link. https://drive.google.com/file/d/1xYsBiCSmXjE0BwoiH_Bcm4AWaA1pESHF/view?usp=sharing.
> > > > > Thank!
> > > > 
> > > > 
> > > > Hi, thank you for the quick response. I have downloaded the new version, but it seems the same as the previous version. Would you please let me know which script is used to extract the features? Thank you.
> > 
> > 
> > Well no... But you can consider trying this [repo](https://github.com/GowthamGottimukkala/I3D_Feature_Extraction_resnet) if you just want the i3d extractor.
> 
> Another question, the original paper used i3d two-stream, but the link is resnet, is the speed and accuracy a little different right?

Which paper are you referring to? For “Real world Anomaly Detection in Surveillance Videos”, I am not very sure. For “Weakly-supervised Video Anomaly Detection with Robust Temporal Feature Magnitude Learning”, they mention somewhere they are using the resnet as the backbone for I3D feature extraction. There may be some difference but won't be so significant I suppose.

**Comment by zimengxueying** [comment]
> 

two stream that is this "[Real-world-Anomaly-Detection-in-Surveillance-Videos-pytorch](https://github.com/seominseok0429/Real-world-Anomaly-Detection-in-Surveillance-Videos-pytorch)",the other    paper you mentioned I have also read. I'm a little confused that the repo you provide, the dimension of the extracted feature is (n/16,2048) right? n is the length of one video, however, this repo provided (32,1024)for rgb and(32,1024)for optical flow."32" is fixed by divided the video by 32 segments even if the length of video is different. But n/16 is not fixed, so that is my question, or I’m misunderstanding, could you correct me? Sorry to bother you.

**Comment by coranholmes** [comment]
> > 
> 
> two stream that is this "[Real-world-Anomaly-Detection-in-Surveillance-Videos-pytorch](https://github.com/seominseok0429/Real-world-Anomaly-Detection-in-Surveillance-Videos-pytorch)",the other paper you mentioned I have also read. I'm a little confused that the repo you provide, the dimension of the extracted feature is (n/16,2048) right? n is the length of one video, however, this repo provided (32,1024)for rgb and(32,1024)for optical flow."32" is fixed by divided the video by 32 segments even if the length of video is different. But n/16 is not fixed, so that is my question, or I’m misunderstanding, could you correct me? Sorry to bother you.

I only read RTFM source codes in details. They have some preprocessing in training phase, so the n/16 segments would be finally transformed to 32. see [here](https://github.com/tianyu0207/RTFM/blob/main/utils.py#L32)

**Comment by zimengxueying** [comment]
> > > 
> > 
> > 
> > two stream that is this "[Real-world-Anomaly-Detection-in-Surveillance-Videos-pytorch](https://github.com/seominseok0429/Real-world-Anomaly-Detection-in-Surveillance-Videos-pytorch)",the other paper you mentioned I have also read. I'm a little confused that the repo you provide, the dimension of the extracted feature is (n/16,2048) right? n is the length of one video, however, this repo provided (32,1024)for rgb and(32,1024)for optical flow."32" is fixed by divided the video by 32 segments even if the length of video is different. But n/16 is not fixed, so that is my question, or I’m misunderstanding, could you correct me? Sorry to bother you.
> 
> I only read RTFM source codes in details. They have some preprocessing in training phase, so the n/16 segments would be finally transformed to 32. see [here](https://github.com/tianyu0207/RTFM/blob/main/utils.py#L32)

Okay, let me read it. Thx for your patience!

**Comment by wennycooper** [comment]
> > > 
> > 
> > 
> > Have you solved the problem? I'm also confused about it. thx
> > > > [@coranholmes](https://github.com/coranholmes) This is the modified download link. https://drive.google.com/file/d/1xYsBiCSmXjE0BwoiH_Bcm4AWaA1pESHF/view?usp=sharing.
> > > > Thank!
> > > 
> > > 
> > > Hi, thank you for the quick response. I have downloaded the new version, but it seems the same as the previous version. Would you please let me know which script is used to extract the features? Thank you.
> 
> Well no... But you can consider trying this [repo](https://github.com/GowthamGottimukkala/I3D_Feature_Extraction_resnet) if you just want the i3d extractor.

Hi, I tried the link you provided. But its output is 2048-dim instead of 1024-dim. Do you have better suggestion?

**Comment by wennycooper** [comment]
> > > > 
> > > 
> > > 
> > > two stream that is this "[Real-world-Anomaly-Detection-in-Surveillance-Videos-pytorch](https://github.com/seominseok0429/Real-world-Anomaly-Detection-in-Surveillance-Videos-pytorch)",the other paper you mentioned I have also read. I'm a little confused that the repo you provide, the dimension of the extracted feature is (n/16,2048) right? n is the length of one video, however, this repo provided (32,1024)for rgb and(32,1024)for optical flow."32" is fixed by divided the video by 32 segments even if the length of video is different. But n/16 is not fixed, so that is my question, or I’m misunderstanding, could you correct me? Sorry to bother you.
> > 
> > 
> > I only read RTFM source codes in details. They have some preprocessing in training phase, so the n/16 segments would be finally transformed to 32. see [here](https://github.com/tianyu0207/RTFM/blob/main/utils.py#L32)
> 
> Okay, let me read it. Thx for your patience!

Did you solve the problem?  I'm looking for pytorch I3D feature extractor which output 1024-dim features.  Any suggestion will be appreciated.
